packages/init/src/SliceMachineInitProcess.ts

```diff
@@ -0,0 +1,1114 @@
+import * as fs from "node:fs/promises";
+import * as path from "node:path";
+
+import chalk from "chalk";
+import { ExecaChildProcess } from "execa";
+import open from "open";
+import logSymbols from "log-symbols";
+import { globby } from "globby";
+
+import {
+	createSliceMachineManager,
+	PrismicUserProfile,
+	PrismicRepository,
+	SliceMachineManager,
+} from "@slicemachine/manager";
+
+import { detectFramework, Framework } from "./lib/framework";
+import {
+	installDependencies,
+	detectPackageManager,
+	PackageManagerAgent,
+	getRunScriptCommand,
+	getExecuteCommand,
+} from "./lib/packageManager";
+import {
+	getRandomRepositoryDomain,
+	formatRepositoryDomain,
+	validateRepositoryDomain,
+	validateRepositoryDomainAndAvailability,
+	getErrorMessageForRepositoryDomainValidation,
+} from "./lib/repositoryDomain";
+import { listr, listrRun } from "./lib/listr";
+import { prompt } from "./lib/prompt";
+import { assertExists } from "./lib/assertExists";
+import { START_SCRIPT_KEY, START_SCRIPT_VALUE } from "./constants";
+
+export type SliceMachineInitProcessOptions = {
+	repository?: string;
+	push?: boolean;
+	pushSlices?: boolean;
+	pushCustomTypes?: boolean;
+	pushDocuments?: boolean;
+	cwd?: string;
+} & Record<string, unknown>;
+
+const DEFAULT_OPTIONS: SliceMachineInitProcessOptions = {
+	push: true,
+	pushSlices: true,
+	pushCustomTypes: true,
+	pushDocuments: true,
+};
+
+export const createSliceMachineInitProcess = (
+	options?: SliceMachineInitProcessOptions,
+): SliceMachineInitProcess => {
+	return new SliceMachineInitProcess(options);
+};
+
+type SliceMachineInitProcessContext = {
+	framework?: Framework;
+	packageManager?: PackageManagerAgent;
+	installProcess?: ExecaChildProcess;
+	userProfile?: PrismicUserProfile;
+	userRepositories?: PrismicRepository[];
+	repository?: {
+		domain: string;
+		exists: boolean;
+	};
+	projectInitialization?: {
+		patchedScript?: boolean;
+	};
+};
+
+export class SliceMachineInitProcess {
+	protected options: SliceMachineInitProcessOptions;
+	protected manager: SliceMachineManager;
+
+	protected context: SliceMachineInitProcessContext;
+
+	constructor(options?: SliceMachineInitProcessOptions) {
+		this.options = { ...DEFAULT_OPTIONS, ...options };
+		this.manager = createSliceMachineManager({ cwd: options?.cwd });
+
+		this.context = {};
+	}
+
+	async run(): Promise<void> {
+		// We prefer to manually allow console logs despite the app being a CLI to catch wild/unwanted console logs better
```

Comment:
makes sense

Reply:
a utility function cold also do this and avoid writing "// eslint-disable-next-line no-console" every where

Reply:
A utility function is less clear and defeats the purpose of the ESLint rule, in my opinion. We want to explicitly tell the reader that we are purposely overriding the ESLint rule. Making it easier to do so could accidentally introduce unwanted `console.log()`s.